RIOT-OS/RIOT
# dist/tools/doccheck: add exclude file for warnings and use it

### Contribution description
This is purely quick-and-lazy fix to finally unblock RIOT-OS/riotdocker#104 (see https://github.com/RIOT-OS/riotdocker/pull/104#issuecomment-905844998), but may also be useful to add future exclude patterns.

The exclude file was generated using (on Arch Linux, 1.9.1)

```sh
make doc 2> dist/tools/doccheck/exclude_patterns_base
grep "^${PWD}" dist/tools/doccheck/exclude_patterns_base | \
    sed -E -e "s#^${PWD}/(.*:)([0-9]+): #\1DIGITS_MAGIC: #" \
           -e 's/[]\\.$*{}|+?()[^-]/\\&/g' \
           -e 's/:DIGITS_MAGIC:/:[0-9]+:/' | \
        sort -u > dist/tools/doccheck/exclude_patterns
rm dist/tools/doccheck/exclude_patterns_base
```

Also, some fixes to the Doxyfile proposed by the newer Doxygen version are piggy-backed.




### Testing procedure
`doccheck` in

```sh
make static-test
```

should pass with newer doxygen versions than the one in the CI. If you are impatient like me, you can comment out the other checks ;-)

<details>

```diff
diff --git a/dist/tools/ci/static_tests.sh b/dist/tools/ci/static_tests.sh
index a04c5a9d40..d836aa7c1a 100755
--- a/dist/tools/ci/static_tests.sh
+++ b/dist/tools/ci/static_tests.sh
@@ -111,25 +111,25 @@ fi
 
 export BASE_BRANCH="${CI_BASE_BRANCH}"
 
-run ./dist/tools/whitespacecheck/check.sh "${BASE_BRANCH}"
-DIFFFILTER="MR" ERROR_EXIT_CODE=0 run ./dist/tools/licenses/check.sh
-DIFFFILTER="AC" run ./dist/tools/licenses/check.sh
+# run ./dist/tools/whitespacecheck/check.sh "${BASE_BRANCH}"
+# DIFFFILTER="MR" ERROR_EXIT_CODE=0 run ./dist/tools/licenses/check.sh
+# DIFFFILTER="AC" run ./dist/tools/licenses/check.sh
 run ./dist/tools/doccheck/check.sh
-run ./dist/tools/externc/check.sh
-run ./dist/tools/cppcheck/check.sh
-run ./dist/tools/vera++/check.sh
-run ./dist/tools/coccinelle/check.sh
-run ./dist/tools/flake8/check.sh
-run ./dist/tools/headerguards/check.sh
-run ./dist/tools/buildsystem_sanity_check/check.sh
-run ./dist/tools/feature_resolution/check.sh
-run ./dist/tools/boards_supported/check.sh
-run ./dist/tools/codespell/check.sh
-if [ -z "${GITHUB_RUN_ID}" ]; then
-    run ./dist/tools/uncrustify/uncrustify.sh --check
-else
-    run ./dist/tools/uncrustify/uncrustify.sh
-fi
-ERROR_EXIT_CODE=0 run ./dist/tools/shellcheck/check.sh
+# run ./dist/tools/externc/check.sh
+# run ./dist/tools/cppcheck/check.sh
+# run ./dist/tools/vera++/check.sh
+# run ./dist/tools/coccinelle/check.sh
+# run ./dist/tools/flake8/check.sh
+# run ./dist/tools/headerguards/check.sh
+# run ./dist/tools/buildsystem_sanity_check/check.sh
+# run ./dist/tools/feature_resolution/check.sh
+# run ./dist/tools/boards_supported/check.sh
+# run ./dist/tools/codespell/check.sh
+# if [ -z "${GITHUB_RUN_ID}" ]; then
+#     run ./dist/tools/uncrustify/uncrustify.sh --check
+# else
+#     run ./dist/tools/uncrustify/uncrustify.sh
+# fi
+# ERROR_EXIT_CODE=0 run ./dist/tools/shellcheck/check.sh
 
 exit $RESULT
```

</details>

Static tests should pass in CI.



### Issues/PRs references
Should unblock https://github.com/RIOT-OS/riotdocker/pull/104
